### PR TITLE
[PY-399][externa] Meta methods & tests for setting layout

### PR DESCRIPTION
**Pull request by JBWilkie**
# Problem
Currently no Meta methods for setting item layout

# Solution
Introducted functions to do this for Meta Items & ItemQuery objects

# Changelog
Meta methods to set the layout of Items & ItemQuery objects

### CODE REVIEW

File: darwin/future/core/items/set_item_layout.py [hunk 1]
```diff
@@ -28,6 +29,12 @@ def set_item_layout(
     Returns:
         JSONType: The response data.
     """
+    if not isinstance(layout, ItemLayout):
+        try:
+            layout = ItemLayout(**layout)
+        except ValueError:
```
Comment (owencjones) [inline darwin/future/core/items/set_item_layout.py:35]: Do these not throw `ValidationError`, potentially as well?

File: darwin/future/meta/queries/item.py [hunk 2]
```diff
@@ -143,3 +144,25 @@ def archive(self) -> None:
         ids = [item.id for item in self]
         filters = {"item_ids": [str(item) for item in ids]}
         archive_list_of_items(self.client, team_slug, dataset_ids, filters)
+
+    def set_layout(self, layout: ItemLayout) -> None:
+        if "team_slug" not in self.meta_params:
```
Comment (owencjones) [inline darwin/future/meta/queries/item.py:149]: I'd maybe space out these individual `if`s a bit, but that's a minor point!

Comment (owencjones) [reply darwin/future/meta/queries/item.py:149]: ...actually a bit surprised that Black didn't tbh

File: darwin/future/tests/meta/objects/test_itemmeta.py [hunk 3]
```diff
@@ -226,3 +226,44 @@ def test_archive_with_bad_team_slug(item: Item) -> None:
     with pytest.raises(AssertionError):
         item.meta_params["team_slug"] = 123
         item.archive()
+
```
Comment (owencjones) [inline darwin/future/tests/meta/objects/test_itemmeta.py:229]: These tests are bang on, nice one

**Review by owencjones — COMMENTED** [review]
Really good stuff John, you're taking previous input into account in new code nicely.

File: darwin/future/meta/queries/item.py (hunk 2 above)
Comment (JBWilkie) [reply darwin/future/meta/queries/item.py:149]: Good point, good for readability

File: darwin/future/core/items/set_item_layout.py (hunk 1 above)
Comment (JBWilkie) [reply darwin/future/core/items/set_item_layout.py:35]: Nice spot, they could indeed

Will except ValidationError as well in these

**Review by Nathanjp91 — APPROVED** [review]

**Pull request merged**